### PR TITLE
Derive traits for newtype aliases

### LINKED ISSUE #1794: Derive standard traits for generated Newtype aliases
### Input C/C++ Header
```C
typedef int foo;
```

### Bindgen Invocation
```Rust
bindgen::Builder::default()
    .header("input.h")
    .default_alias_style(bindgen::AliasVariation::NewType)
    .generate()
    .unwrap()
```

### Actual Results
```rust
#[repr(transparent)]
pub struct foo(pub ::std::os::raw::c_int);
```

### Expected Results
```rust
#[derive(Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
#[repr(transparent)]
pub struct foo(pub ::std::os::raw::c_int);
```

### PR DESCRIPTION
**Pull request by eggyal**
Fixes #1794

### CODE REVIEW

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -793,8 +793,64 @@ impl CodeGenerator for Type {
...
+                        let mut derives = vec![];
+
+                        if item.can_derive_debug(ctx) {
+                            derives.push("Debug");
+                        }
+
+                        if item.can_derive_default(ctx) {
+                            derives.push("Default");
+                        }
+
```
Comment (emilio) [inline src/codegen/mod.rs:807]: Is there any chance we can share this code with the other code that build similar attribute lists?

Comment (eggyal) [reply src/codegen/mod.rs]: Sure—see latest commit.

**Comment by eggyal** [comment]
Seems that Travis failed because of rustfmt?  How do I request it try again?

**Comment by emilio** [comment]
You can rebase on top of master, which should have that fixed.

File: src/codegen/mod.rs [hunk 2]
```diff
@@ -95,6 +95,93 @@ fn root_import(
...
+            (DerivableTraits::COPY, "Copy"),
+            (DerivableTraits::CLONE, "Clone"),
+            (DerivableTraits::HASH, "Hash"),
+            (DerivableTraits::PARTIAL_ORD, "PartialOrd"),
+            (DerivableTraits::ORD, "Ord"),
+            (DerivableTraits::PARTIAL_EQ, "PartialEq"),
+            (DerivableTraits::EQ, "Eq"),
+        ]
+        .iter()
+        .filter_map(|&(flag, derive)| {
```
Comment (emilio) [inline src/codegen/mod.rs:178]: I'd mildly prefer to just do:

```
if !derivable_traits.contains(flag) {
  return None;
}
Some(derive)
```

But no big deal.

**Pull request merged**